discovery_engine/lib/src/discovery_engine_base.dart

```diff
@@ -291,7 +291,7 @@ class DiscoveryEngine {
       logger.e(message, error, stackTrace);
 
       // into [EngineExceptionRaised] event with a specific reason
-      return EngineEvent.engineExceptionRaised(reason);
+      return EngineEvent.engineExceptionRaised(reason, error.toString());
```

Comment:
this will do the string conversion
```suggestion
      return EngineEvent.engineExceptionRaised(reason, '$error');
```